tests/integration/loading/test_loading_component.py

```diff
@@ -0,0 +1,268 @@
+
+    assert not dash_dcc.get_logs()
+
+
+def test_ldcp005_dynamic_loading_component(dash_dcc):
+    lock = Lock()
+
+    app = dash.Dash(__name__, suppress_callback_exceptions=True)
+
+    app.layout = html.Div([html.Button(id="btn-1"), html.Div(id="div-1")])
+
+    @app.callback(Output("div-1", "children"), [Input("btn-1", "n_clicks")])
+    def updateDiv(n_clicks):
+        if n_clicks is None:
+            return
+
+        with lock:
+            return html.Div(
+                [
+                    html.Button(id="btn-2"),
+                    dcc.Loading([html.Button(id="btn-3")], className="loading-1"),
+                ]
+            )
+
+    @app.callback(Output("btn-3", "children"), [Input("btn-2", "n_clicks")])
+    def updateDynamic(n_clicks):
+        if n_clicks is None:
+            return "content"
+
+        with lock:
+            return "changed"
+
+    dash_dcc.start_server(app)
+
+    dash_dcc.find_element("#btn-1")
+    dash_dcc.wait_for_text_to_equal("#div-1", "")
+
+    dash_dcc.find_element("#btn-1").click()
+
+    dash_dcc.find_element("#div-1 #btn-2")
+    dash_dcc.wait_for_text_to_equal("#btn-3", "content")
+
+    with lock:
+        dash_dcc.find_element("#btn-2").click()
+
+        dash_dcc.find_element(".loading-1 .dash-spinner")
+        dash_dcc.wait_for_text_to_equal("#btn-3", "")
+
+    dash_dcc.wait_for_text_to_equal("#btn-3", "changed")
+
+    assert not dash_dcc.get_logs()
+
+
+def test_ldcp006_children_identity(dash_dcc):
+    lock = Lock()
+
+    app = dash.Dash(__name__)
+    app.layout = html.Div([
+        html.Button("click", id="btn"),
+        dcc.Loading(dcc.Graph(id="graph"), className="loading")
+    ])
+
+    @app.callback(Output("graph", "figure"), [Input("btn", "n_clicks")])
+    def update_graph(n):
+        with lock:
+            bars = list(range(2, (n or 0) + 5))
+            return {
+                "data": [{"type": "bar", "x": bars, "y": bars}],
+                "layout": {"width": 400, "height": 400}
+            }
+
+    with lock:
+        dash_dcc.start_server(app)
+        dash_dcc.find_element(".loading .dash-spinner")
+        dash_dcc.find_element("#graph .js-plotly-plot")
+        dash_dcc.driver.execute_script(
+            "window.gd = document.querySelector('.js-plotly-plot');"
+            "window.gd.__test__ = 'boo';"
+        )
+
+    test_identity = (
+        "var gd_ = document.querySelector('.js-plotly-plot');"
+        "return gd_ === window.gd && gd_.__test__ === 'boo';"
+    )
+
+    assert len(dash_dcc.find_elements('.js-plotly-plot .bars path')) == 3
+    assert dash_dcc.driver.execute_script(test_identity)
+
+    with lock:
+        dash_dcc.find_element("#btn").click()
+        dash_dcc.find_element(".loading .dash-spinner")
+        assert len(dash_dcc.find_elements('.js-plotly-plot .bars path')) == 3
+        assert dash_dcc.driver.execute_script(test_identity)
+
+    assert len(dash_dcc.find_elements('.js-plotly-plot .bars path')) == 4
```

Comment:
To go along with the graph not being visible while loading, can we make sure it is visible now.

Reply:
added `getComputedStyle` visibility tests in https://github.com/plotly/dash-core-components/pull/740/commits/4fc809a6c1328f922c0eddff315adc1bf79bd680